src/compas_timber/_fabrication/jack_cut.py

```diff
@@ -0,0 +1,274 @@
+        self._start_x = start_x
+
+    @property
+    def start_y(self):
+        return self._start_y
+
+    @start_y.setter
+    def start_y(self, start_y):
+        if start_y > 50000.0:
+            raise ValueError("Start Y must be less than 50000.0.")
+        self._start_y = start_y
+
+    @property
+    def start_depth(self):
+        return self._start_depth
+
+    @start_depth.setter
+    def start_depth(self, start_depth):
+        if start_depth > 50000.0:
+            raise ValueError("Start Depth must be less than 50000.0.")
+        self._start_depth = start_depth
+
+    @property
+    def angle(self):
+        return self._angle
+
+    @angle.setter
+    def angle(self, angle):
+        if angle > 179.9 or angle < 0.1:
+            raise ValueError("Angle must be between 0.1 and 179.9.")
+        self._angle = angle
+
+    @property
+    def inclination(self):
+        return self._inclination
+
+    @inclination.setter
+    def inclination(self, inclination):
+        if inclination > 179.9 or inclination < 0.1:
+            raise ValueError("Inclination must be between 0.1 and 179.9.")
+        self._inclination = inclination
+
+    ########################################################################
+    # Alternative constructors
+    ########################################################################
+
+    @classmethod
+    def from_plane_and_beam(cls, plane, beam, ref_side_index=0):
+        """Create a JackRafterCut instance from a cutting plane and the beam it should cut.
+
+        Parameters
+        ----------
+        plane : :class:`~compas.geometry.Plane` or :class:`~compas.geometry.Frame`
+            The cutting plane.
+        beam : :class:`~compas_timber.elements.Beam`
+            The beam that is cut by this instance.
+        ref_side_index : int, optional
+            The reference side index of the beam to be cut. Default is 0 (i.e. RS1).
+
+        Returns
+        -------
+        :class:`~compas_timber.fabrication.JackRafterCut`
+
+        """
+        # type: (Plane | Frame, Beam, int) -> JackRafterCut
+        if isinstance(plane, Frame):
+            plane = Plane.from_frame(plane)
+        start_y = 0.0
+        start_depth = 0.0
+        ref_side = beam.ref_sides[ref_side_index]  # TODO: is this arbitrary?
+        ref_edge = Line.from_point_and_vector(ref_side.point, ref_side.xaxis)
+        orientation = cls._calculate_orientation(ref_side, plane)
+
+        point_start_x = intersection_line_plane(ref_edge, plane)
+        if point_start_x is None:
+            raise ValueError("Plane does not intersect with beam.")
+
+        start_x = distance_point_point(ref_edge.point, point_start_x)
+        angle = cls._calculate_angle(ref_side, plane, orientation)
+        inclination = cls._calculate_inclination(ref_side, plane, orientation)
+        return cls(orientation, start_x, start_y, start_depth, angle, inclination, ref_side_index=ref_side_index)
```

Comment:
should the ref_side_index be in the this class constructor, since it is not a parameter of the class?

Reply:
`ref_side_index` is an attribute of the parent class, as all processes need this parameter. notice `JackRafterCut` knows nothing about it, it gets it as a `kwarg` and passes it through to the parent in its call to `super().__init__()`.